fog/ingest/server/src/controller.rs

```diff
@@ -336,10 +336,13 @@ where
             // This also returns an ingress key state which indicates if publishing was
             // successful, it may fail if the key is retired. Then we can check
             // if we are still needed to be active.
-            if state.is_active() {
+            if state.is_active() && state.is_last_report_published_block_index(block.index) {
```

Comment:
Sarah mentioned this too...We thought maybe `has_report_expired()` is a better name?  Maybe `is_report_expired()`?   To answer your question further down, this function is supposed to check to see if the current block.index has reached the pubkey expiry window.  If it has, then when we successfully publish the report and update the variable to store the block index of the last report published.

When looking further into it though, I think the logic for the `has_report_expired()` function is incorrect.  I think we actually need the absolute value of  `pubkey_expiry_window - block.index`, so that would mean the function should be this:
(I didnt find a Rust abs() function on u64 so this is implementing that)
```
pub fn has_expired_report(&self, block_index: u64) -> bool {
        let compare : u64;
        if block_index < self.pubkey_expiry_window {
            compare = self.pubkey_expiry_window.saturating_sub(block_index);
        } else {
            compare = block_index.saturating_sub(self.pubkey_expiry_window);
        }
        self.last_report_published_block_index == compare
    }
```

Reply:
I think there is a potentially more elegant solution. Since we now keep track of what is the last pubkey expiry we published, what would happen if `publish_report` would change to:
1) not publish a new report if the calculated pubkey expiry is lower than the last published one
2) be in charge of updating `state.last_report_published_block_index` if a new report has been successfully published
It seems to be that this will both simplify the logic, and remove the need to do separate book-keeping using `set_last_report_published_block_index` since I don't see when we would want to call that without previously calling `publish_report`.